src/stdfblib/ita/debug_device/DebugDevice.h

```diff
@@ -0,0 +1,34 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Jose Cabral
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Jose Cabral - initial implementation
+ *******************************************************************************/
+
+#pragma once
+
+#include "RMT_DEV.h"
+#include "DebugMGR.h"
+
+/**
+ * @brief Device that adds debug commands to the device. The commands are defined in DebugMGR
+ * 
+ */
+class DebugDevice : public RMT_DEV {
```

Comment:
why remote device when you use the OPC UA DEV_MGR?

Reply:
The OPCUA_DEV is not offering any extra inputs or capabilities than the ones offered by the `OPCUA_MGR`, so I found it better to have the OPCUA_MGR as a member instead of getting it via inheritance. Also, if one would want to add even more capbilities on top of debugging (something I'm currently working on, that's when I realized this seemed a better choice), one would need to inherit again from `DebugDevice`, which is not a good practice.

Reply:
Sorry I just noticed my comment was way to short. I totally agree with you that aggregation is much better then in inheritance. What I wanted to say is that I don't think you should inherit your DebugDevice from RM_DEV and also not from OPCUA_DEV. I meant just from CDevice. This would be the cleanest and most correct solution.